### PR TITLE
Release N50

### LINKED ISSUE #1474: Newsletter 50: April 2024
# Newsletter 50: April 2024

**Editors:** @janhohenheim, @AngelOnFira, and @mamaicode 

Another month has gone by, so it's time to put together the Rust Gamedev newsletter with April's news!

## Current Schedule

The deadline for all section PRs is the **28th of April, 2024**

Any PRs submitted after this date will be moved to the next newsletter period. The target release date for this newsletter is the 3rd of May, 2024

## Current Structure & Status

Below is our current planned structure for the newsletter, and the status of each PR (which we'll try to keep updated).

This is **not** an exhaustive list - if you have your own project that you want to write about, just make a comment on this issue and open a PR!

### Game Updates
- [x] sm64jsarchive by @uuphoria2: https://github.com/rust-gamedev/rust-gamedev.github.io/pull/1477
- [x] OpenCombat by @buxx 
- [x] Times of Progress by @wilk10 
- [x] Jumpy Steam Playtest by @erlend-sh / @MaxCWhitehead
- [X] Monk Tower @maciekglowka 
- [X] WoR by @MasonRemaley 
- [X] You Are Merlin by @hseager 

### Learning Material Updates

- [X] Building Android Games by @maciekglowka 

### Engine Updates

- [x] Bottomless Pit by @EggShark
- [x] Bevy 0.13.2 by @janhohenheim (or, if they want, @alice-i-cecile / @mockersf)

### Tooling Updates

None yet. Feel free to submit yours!

### Library Updates

- [x] Lightyear by @cBournhonesque 
- [x] Hexx by @ManevilleF (@janhohenheim will do this part if Felix doesn't find the time)
- [x] [joltc_sys](https://docs.rs/joltc-sys/0.1.1+Jolt-5.0.0/joltc_sys/) / [rolt](https://docs.rs/rolt/0.1.1+Jolt-5.0.0/rolt/) by @LPGhatguy?

### Other News

- [x] [Chris Biscardi - Bevy: A case study in ergonomic Rust](https://www.youtube.com/watch?v=CnoDOc6ML0Y) by @ChristopherBiscardi
- [x] [Reactivity in Bevy](https://machinewords.hashnode.dev/reactivity-in-bevy-from-the-bottom-up-part-1) by @viridia?

### Discussions

None yet. Feel free to submit yours!

## Publishing Steps

- [ ] Final review - by everyone
- [ ] Publish - by @janhohenheim
- [ ] Post on /r/rust, /r/rust_gamedev, /r/gamedev, URLO - by @janhohenheim
- [ ] Pin thread on Twitter - by @janhohenheim
- [ ] Add comment links - by @janhohenheim

## How to Contribute

If you want to help writing the newsletter:

- **Read [CONTRIBUTING.md](https://github.com/rust-gamedev/rust-gamedev.github.io/blob/source/CONTRIBUTING.md?rgh-link-date=2023-09-12T16%3A17%3A34Z).**
- Choose one or more of the "🆓 **free**" sections listed below, and leave a comment letting us know you want to work on them.
  - The links in brackets (like `[1](#), [2](#), [3](#)`) are suggestions of links to include in the section. Feel free to add more!
  - The username listed next to the section (like `@janhohenheim?`) is a suggestion of who may want to pick up the work (usually the project's developer, or someone who has expressed interest in the past).
  - You are not obligated to write a section if you're tagged or your project is listed! You're welcome to ask someone else to write the section, or to ask for your project to be excluded from this month's post.
  - Extra sections not listed in the plan are welcomed - just leave a comment and open a PR!
- Write a short overview in the newsletter's Markdown file, making sure to follow the style guidelines (see below).
- Send a PR _to the `source` branch_ (example: [N15: A/B Street #336](https://github.com/rust-gamedev/rust-gamedev.github.io/pull/336)).
- Mention this issue in your PR's description to link it all together.

## Style Guidelines

The full style guide is in [CONTRIBUTING.md](https://github.com/rust-gamedev/rust-gamedev.github.io/blob/source/CONTRIBUTING.md?rgh-link-date=2023-09-12T16%3A17%3A34Z),
but here are the most important rules:

- Write in third-person perspective.
- Each line must be 80 characters or less, for ease of reviewing/diffing.
- Only one image per section is allowed.
  - The maximum size is 300kb for static images and 2.5mb for GIFs.
  - The image should come before the text, and must have alt text for accessibility.
  - Prefer static images to GIFs, to keep the page load times down.
- Each section should be under 1000 characters, and under 6 paragraphs.
  - This only applies to the rendered text, not the markup.
- Keep formatting minimal - no bold/italics/etc.
- Avoid long/nested bullet point lists - no changelogs!

Please use these templates as a starting point:

**Games/apps/libraries:**

```md
### [Game name]

![alt text](img)
_optional image label_

[Game name] ([GitHub], [Discord], [Twitter]) by [@nickname]
is... {short project description in one sentence}.

{An overview of the recent updates with links to more details}.

_Discussions: [/r/rust_gamedev](link), [Twitter](link), [etc](link)_

[Game name]: http://example.com
```

**Articles/blog posts/videos/etc:**

```md
### [Article name]

![alt text](img)
_optional image label_

[@nickname] published an [article] about...
{overview what the resource is about}.

_Discussions: [/r/rust_gamedev](link), [Twiter](link), [etc](link)_

[Article name]: http://example.com
```

### PR DESCRIPTION
**Pull request by janhohenheim**
Resolves #1474

Scheduled for **2024-05-03**

### CODE REVIEW

File: content/news/050/index.md
Comment (Vrixyz) [inline content/news/050/index.md:126]: I think it's fine to have "inline links", but as historically **ALL** links were defined with a label I'm just noting this to start the discussion, just so we're clear about the expectations.

File: content/news/050/index.md
Comment (Vrixyz) [inline content/news/050/index.md:328]: a todo (or toremove?)

File: content/news/050/index.md
Comment (janhohenheim) [reply content/news/050/index.md:126]: Yeah, I noticed that too. Agreed, it's good to point this out.
I removed the lints for that intentionally because it doesn't matter for markdown that is not maintained after the post goes up. I can change it back if other maintainers disagree, but I'd prefer to be relaxed about this.

File: content/news/050/index.md
Comment (janhohenheim) [reply content/news/050/index.md:328]: Still working on it :)

File: content/news/050/index.md
Comment (Vrixyz) [inline content/news/050/index.md:368]:
```suggestion
LogLog games has published a [very well-written blog post](https://loglog.games/blog/leaving-rust-gamedev/>) about their reasons to leave Rust gamedev.
```

it's less strong than their "decision", but still the article is clearly written on their particular perspective.

**Comment by janhohenheim** [comment]
Initial version is done; Feedback welcome :)

**Review by Vrixyz — APPROVED** [review]

**Review by erlend-sh — APPROVED** [review]

**Review by AngelOnFira — CHANGES_REQUESTED** [review]
blocking to hold merge on my review (expected in <24h) 💯

**Review by AngelOnFira — CHANGES_REQUESTED** [review]
I made a separate PR into this branch with many changes, that should be merged 👍🏻 https://github.com/rust-gamedev/rust-gamedev.github.io/pull/1501

File: content/news/050/index.md
Comment (AngelOnFira) [inline content/news/050/index.md:249]: I think it's worth leaving in version number, since it suggests a new version was released, instead of just general progress on a project. For example, I imagine we'd have a Bevy release's section titled `Bevy 0.XX` to signify the importance.

File: content/news/050/index.md
Comment (AngelOnFira) [inline content/news/050/index.md:110]: nit: wordy way of saying the same thing I think?

update: seems like this wording is because items are being removed (addressing that in another comment). I think staying on "Recent updates" makes more sense still.

File: content/news/050/index.md
Comment (AngelOnFira) [inline content/news/050/index.md:115]: Instead of removing this list, can we turn it into a paragraph if it comes out too long? It isn't great removing someone's content after it was merged. Will make a suggestion comment.

Comment (AngelOnFira) [reply content/news/050/index.md:115]: Actually there isn't a great way for me to suggest over this whole block, so I'll just post it here:
```
Recently, a [closed beta][wor-closed-beta] began. All characters now have
voices, and various [speedrunning features][wor-speedrun] were added. A [reset
button][wor-reset] was added for when undo is inconvenient. Speech bubbles now
[identify the speaker][wor-speech-bubble], and orbs are now shown with an "xray"
effect so they are visible even when behind scenery. Older CPUs without AVX
support are [now supported][wor-avx]. Secrets were made harder, and an [in-game
achievement UI][wor-achievements] was created. An [in-game Jukebox
UI][wor-jukebox] was also created.

For the full changelog, see the [release notes][wor-release-notes].
```

File: content/news/050/index.md
Comment (AngelOnFira) [inline content/news/050/index.md:105]: I think Rust isn't necessarily implied here, since some projects might only use Rust for part of the game. I think it would be good to keep the specifics in.

```suggestion
mechanics being developed by [@masonremaley] in a custom Rust engine.
```

File: content/news/050/index.md
Comment (AngelOnFira) [inline content/news/050/index.md:207]:
```suggestion
Players can choose their favourite visual theme such as Zelda, Warcraft, and Fallout. There is also mobile and web support.
```

**Review by AngelOnFira — CHANGES_REQUESTED** [review]
Some additional items to look at

File: content/news/050/index.md
Comment (janhohenheim) [reply content/news/050/index.md:95]: I intentionally removed some content after feedback that the newsletter contained too many details most people don't care about. Personally, as a contributor, I would expect the editors to chop up my content however they think is best before release.

File: content/news/050/index.md
Comment (janhohenheim) [reply content/news/050/index.md:187]: Misleading: there is only CLI and web. The web version supports mobile devices.

File: content/news/050/index.md
Comment (janhohenheim) [reply content/news/050/index.md:249]: Good point, but then I'd like to omit the patch version, as otherwise it seems like we will be talking about some minor bugfixes only. As far as I can tell, this entry concerns `0.13.0` and `0.13.1`, so I'll write it as `0.13`

**Comment by janhohenheim** [comment]
@Vrixyz I know you've distanced yourself in the past from phrases like "Jan and Thierry have led the newsletter revival". Could you check the wording in the announcement after @AngelOnFira's changes?

**Comment by Vrixyz** [comment]
@janhohenheim I feel like you deserve more credits in the wording but I'm fine with it 👍

File: content/news/050/index.md
Comment (janhohenheim) [reply content/news/050/index.md]: [`44c261b` (#1496)](https://github.com/rust-gamedev/rust-gamedev.github.io/pull/1496/commits/44c261b5cd65f664854a03a9ead0951a1d145f9a)

File: content/news/050/index.md
Comment (janhohenheim) [reply content/news/050/index.md:249]: [`44c261b` (#1496)](https://github.com/rust-gamedev/rust-gamedev.github.io/pull/1496/commits/44c261b5cd65f664854a03a9ead0951a1d145f9a)

File: content/news/050/index.md
Comment (janhohenheim) [reply content/news/050/index.md]: Counteroffer: [`5fa2fed` (#1496)](https://github.com/rust-gamedev/rust-gamedev.github.io/pull/1496/commits/5fa2fed7dfbe48aee35e083387b3ba4a6a10fa8e)

**Comment by janhohenheim** [comment]
> @janhohenheim I feel like you deserve more credits in the wording but I'm fine with it 👍

Then I'll leave it to @AngelOnFira on how to phrase it. I don't want to toot my own horn too much, haha.

File: content/news/050/index.md
Comment (AngelOnFira) [reply content/news/050/index.md]: Looks good, but can we also add the comma after "Warcraft"?

**Comment by AngelOnFira** [comment]
> I know you've distanced yourself in the past from phrases like "Jan and Thierry have led the newsletter revival"

In rewording that paragraph, one item I tried to emphasize is that there still is a *team* behind the newsletter, it's not just a single person. I'm a bit adverse to giving off the impression that a single person is running everything without support, while also stating that the previous issue hiatus was a result of maintainer burnout 😅 if needed, I can try rewording the sentence in a different way.

In regards of credit, I think you both have done quite a lot in getting things rolling, so I'm happy with both names mentioned 👍🏻

File: content/news/050/index.md
Comment (AngelOnFira) [reply content/news/050/index.md]: re: community feedback, is that part of what's coming in from the survey, or was it discussed somewhere else?

I think this is fair though, the new paragraph looks good. It would be good for us to do a pass over instructions of writing sections for the newsletter so that any feedback is incorporated, once that all comes out 💯

File: content/news/050/index.md
Comment (AngelOnFira) [reply content/news/050/index.md]: Actually I added a few others as well, should be easier to merge from a PR https://github.com/rust-gamedev/rust-gamedev.github.io/pull/1503

**Comment by janhohenheim** [comment]
> re: community feedback, is that part of what's coming in from the survey, or was it discussed somewhere else?

From discussions on Discord. I don't like calling out specific names, but some very active community members wrote this, which makes me believe their opinion is representative of the community. Of course, if the survey tells us something different, I'd change my stance accordingly

**Review by AngelOnFira — APPROVED** [review]
Awesome! Everything looks good to me 🦀

**Pull request merged**
